### ISSUE TITLE
haskell-ci 0.19.20241111 silently discards GHC 8.4.4

### ISSUE THREAD

**Issue by andreasabel** [haskell-CI/haskell-ci#750]
I am seeing
> *INFO* Generating GitHub config for testing for GHC versions: 8.0.2 8.2.2 8.4.4 8.6.5 8.8.4 8.10.7 9.0.2 9.2.8 9.4.8 9.6.6 9.8.2 9.10.1

but there is no matrix entry created for ghc-8.4.4.

Generated matrix is:
```yaml
      matrix:
        include:
          - compiler: ghc-9.10.1
            compilerKind: ghc
            compilerVersion: 9.10.1
            setup-method: ghcup
            allow-failure: false
          - compiler: ghc-9.8.2
            compilerKind: ghc
            compilerVersion: 9.8.2
            setup-method: ghcup
            allow-failure: false
          - compiler: ghc-9.6.6
            compilerKind: ghc
            compilerVersion: 9.6.6
            setup-method: ghcup
            allow-failure: false
          - compiler: ghc-9.4.8
            compilerKind: ghc
            compilerVersion: 9.4.8
            setup-method: ghcup
            allow-failure: false
          - compiler: ghc-9.2.8
            compilerKind: ghc
            compilerVersion: 9.2.8
            setup-method: ghcup
            allow-failure: false
          - compiler: ghc-9.0.2
            compilerKind: ghc
            compilerVersion: 9.0.2
            setup-method: ghcup
            allow-failure: false
          - compiler: ghc-8.10.7
            compilerKind: ghc
            compilerVersion: 8.10.7
            setup-method: ghcup
            allow-failure: false
          - compiler: ghc-8.8.4
            compilerKind: ghc
            compilerVersion: 8.8.4
            setup-method: ghcup
            allow-failure: false
          - compiler: ghc-8.6.5
            compilerKind: ghc
            compilerVersion: 8.6.5
            setup-method: ghcup
            allow-failure: false
          - compiler: ghc-8.2.2
            compilerKind: ghc
            compilerVersion: 8.2.2
            setup-method: hvr-ppa
            allow-failure: false
          - compiler: ghc-8.0.2
            compilerKind: ghc
            compilerVersion: 8.0.2
            setup-method: hvr-ppa
            allow-failure: false
```

Setting was:
```
tested-with:
   GHC == 9.10.1
   GHC == 9.8.2
   GHC == 9.6.6
   GHC == 9.4.8
   GHC == 9.2.8
   GHC == 9.0.2
   GHC == 8.10.7
   GHC == 8.8.4
   GHC == 8.6.5
   GHC == 8.4.4
   GHC == 8.2.2
   GHC == 8.0.2
```

**Comment by phadej** [comment]
Adjust jobs, `ghcup-jobs` etc.

**Comment by andreasabel** [comment]
Not sure what you mean, maybe your answer is to fragmentary.
I am working in a setting (almost) without a `cabal.haskell-ci` (it only has `branches: master`).
So the only configuration comes from the `tested-with` field in the `.cabal` file.

Here is the original workflow run: https://github.com/andreasabel/patience/actions/runs/11815323476?pr=1
I used my fork (based on haskell-ci 0.19.20241111) to generate the workflow, but then confirmed that 8.4.4 is also omitted in the original haskell-ci 0.19.20241111.
I now reverted my fork to skip the most recent commits that seemingly introduced the problem:
- #749 

So this PR seemed to have introduced the regression.
